server/sonar-db-dao/src/main/java/org/sonar/db/ce/CeQueueDao.java

```diff
@@ -84,6 +85,10 @@ public int countByQuery(DbSession dbSession, CeTaskQuery query) {
     return mapper(dbSession).selectPendingByMinimumExecutionCount(minExecutionCount);
   }
 
+  public void resetTasksWithUnknownWorkerUUIDs(DbSession dbSession, Set<String> knownWorkerUUIDs) {
+    mapper(dbSession).resetTasksWithUnknownWorkerUUIDs(knownWorkerUUIDs, system2.now());
```

Comment:
the point is not to leverage huge collection as parameter (well, not only), but to ensure no invalid SQL can be generated. Argument is used to generate an IN clause, we should prevent any error. Some SGBD do not support more than a given number of values in the IN clause.

You can't make assumption on the number of nodes and workers a customer may configure. Customers are very imaginative and like to push things to their limit :)

There is no need to change the input of `resetTasksWithUnknownWorkerUUIDs`, `executeLargeUpdates` can be safely modified to take a `Collection` rather than a `List`. If you have a look at the implementation, it's actually faster to provide a `Set` as input